src/arcaflow_plugin_sdk/schema.py

```diff
@@ -5288,6 +5330,48 @@ def __init__(
         self._t = t
         self._scope = scope
         self.discriminator_field_name = discriminator_field_name
+        self.discriminator_inlined = discriminator_inlined
+
+    def _validate_discriminator(self):
+        # You can only validate a schema's compatibility after your
+        # scope has been constructed and populated with all referenced
+        # schemas.
+        for discriminator_value, set_member in self.types.items():
+            if (
+                not self.discriminator_inlined
+                and hasattr(set_member, "properties")
+                and self.discriminator_field_name in set_member.properties
+            ):
+                raise BadArgumentException(
+                    f'object id "{set_member.id}" has conflicting field '
+                    f'"{self.discriminator_field_name}"; either remove that '
+                    "field or set inline to true for "
+                    f"{type(self).__name__}[{type(set_member).__name__}]"
+                )
+            if (
+                self.discriminator_inlined
+                and hasattr(set_member, "properties")
+                and self.discriminator_field_name not in set_member.properties
+            ):
+                raise BadArgumentException(
+                    f'object id "{set_member.id}" needs discriminator field '
+                    f'"{self.discriminator_field_name}"; either add that '
+                    "field or set inline to false for "
+                    f"{type(self).__name__}[{type(set_member).__name__}]"
+                )
+            elif (
+                self.discriminator_inlined
+                and hasattr(set_member, "properties")
+                and self.discriminator_field_name in set_member.properties
+                and not isinstance(discriminator_value, self._t)
+            ):
+                raise BadArgumentException(
+                    f"the type of object id {set_member.id}'s discriminator"
+                    f' field "{self.discriminator_field_name}" does not match'
+                    " OneOfSchema discriminator type; expected"
```

Comment:
Does `OneOfSchema` exist?  Or, should that be `OneOfIntSchema`?

Reply:
It seemed like the best option as opposed to `OneOf*Schema`. Is that what you would propose?

Reply:
did it